activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -53,7 +53,7 @@ def custom_primary_key?
 
       def defined_for?(options_or_to_table = {})
         if options_or_to_table.is_a?(Hash)
-          options_or_to_table.all? {|key, value| options[key].to_s == value.to_s }
+          options_or_to_table.all? {|key, value| send(key).to_s == value.to_s }
```

Comment:
What does this fix? Why does it fix it? This change feels unrelated to the intent of the existing code. Can you give more context as to why this fixes the root problem and isn't a bandaid?